README.md

```diff
@@ -21,7 +21,7 @@ extern crate getset;
 pub struct Foo<T> where T: Copy + Clone + Default {
     /// Doc comments are supported!
     /// Multiline, even.
-    #[get] #[set] #[get_mut]
+    #[get] #[deref] #[set] #[get_mut]
```

Comment:
Can I ask why you chose this option over something like `#[get ="deref"]` or `#[get = "pub deref"]`?

Reply:
It was mostly to be coherent with the current implementation. The string associated with the attribute was the "visibility" of the function, so it was a bit strange to add the `deref` part there. I can change it, of course, but it was strange for me.

Also, if we used the `deref` in that place, would we allow `#[get="deref pub"]`? or only `#[get="pub deref"]`. What about complex visibility modifiers such as `#[get="pub(crate)"]`?

It can be done, but i think it's a bit more cumbersome to define.

Reply:
Yeah. I'm not entirely sure I favor either way. One bother for me is that `#[get_mut] #[deref]` seems odd. It makes `#[deref]` into a no-op. Also `#[deref]` alone would be a no-op.

That said, I could see a legitimate use case under #21 .

Reply:
So, I thought about this more. I think we should take a conservative approach.

The goal of this library is to reduce the boilerplate associated with *trivial* getters and setters in a minimal fashion to support library evolution in a practical manner.

Notably: Field Access is not ABI stable across struct versions. Having Getters and Setters helps with stability.

Thankfully visibility modifiers are fairly easily tokenizable. We can create a simple set from the tokens present, and if `deref` (which is not a valid visibility modifer) is present we can do this deref implementation.

How does that sound to you?

I think ideal would be something like `#[get(vis = "pub", deref)]` but at least when I tried a when I first wrote this it didn't seem possible.